### PR TITLE
Add mi workflow & template & config

### PR DESCRIPTION
**Pull request by xtuchyna**

### CODE REVIEW

File: mi-scheduler/base/argo-workflows/mi-workflow-template.yaml [hunk 1]
```diff
@@ -0,0 +1,141 @@
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: mi
+  annotations:
+    thoth-station.ninja/template-version: 0.1.0
```
Comment (harshad16) [inline mi-scheduler/base/argo-workflows/mi-workflow-template.yaml:6]: please remove this, argocd handles version for manifest files.

File: mi-scheduler/base/argo-workflows/mi-workflow-template.yaml [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      inputs:
+        parameters:
+          - name: REPOSITORY
+          - name: IMAGE_STREAM_REGISTRY
+          - name: IMAGE_STREAM_NAMESPACE
+          - name: IMAGE_STREAM_TAG
+      container:
+        image: "{{inputs.parameters.IMAGE_STREAM_REGISTRY}}/\
+        {{inputs.parameters.IMAGE_STREAM_NAMESPACE}}/\
+        srcopsmetrics:{{inputs.parameters.IMAGE_STREAM_TAG}}"
```
Comment (harshad16) [inline mi-scheduler/base/argo-workflows/mi-workflow-template.yaml:59]:
```suggestion
        image: "mi:latest"
```

Comment (harshad16) [reply mi-scheduler/base/argo-workflows/mi-workflow-template.yaml:59]: As the image are no longer kept in different namespace, we dont need to provide, image_registry, image_namespace,etc. please remove those variables in the workflow.

File: mi-scheduler/base/argo-workflows/mi-workflow-template.yaml [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
```
Comment (harshad16) [inline mi-scheduler/base/argo-workflows/mi-workflow-template.yaml:106]:
```suggestion
        image: "mi:latest"
```

File: mi-scheduler/base/openshift-templates/mi-run-template.yaml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    displayName: ImageStream Project
+    required: true
+    name: IMAGE_STREAM_NAMESPACE
+    value: "thoth-test-core"
+
+  - description: Tag of the ImageStream to be use
+    displayName: ImageStream Tag
+    required: true
+    name: IMAGE_STREAM_TAG
+    value: "latest"
```
Comment (pacospace) [inline mi-scheduler/base/openshift-templates/mi-run-template.yaml:38]: not required anymore :)

File: mi-scheduler/base/argo-workflows/mi-workflow-template.yaml (hunk 2 above)
Comment (pacospace) [inline mi-scheduler/base/argo-workflows/mi-workflow-template.yaml:59]: `image: mi`

File: mi-scheduler/base/openshift-templates/mi-run-template.yaml [hunk 5]
```diff
@@ -0,0 +1,73 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: mi
+  annotations:
+    description: "Meta-information Indicators"
+    openshift.io/display-name: "mi"
+    version: 0.1.0
```
Comment (harshad16) [inline mi-scheduler/base/openshift-templates/mi-run-template.yaml:8]: please remove this , argocd will take care of versioning

File: mi-scheduler/base/configmaps.yaml [hunk 6]
```diff
@@ -2,7 +2,16 @@
...
   organizations: "thoth-station,AICoE"
+---
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: mi
+data:
+  bucket: DH-DEV-DATA
+  host: https://s3.upshift.redhat.com
+  prefix: data/thoth/dtuchyna
```
Comment (pacospace) [inline mi-scheduler/base/configmaps.yaml:17]: there is a `ceph` CM in core to use to store data

File: mi-scheduler/base/openshift-templates/mi-run-template.yaml (hunk 4 above)
Comment (harshad16) [inline mi-scheduler/base/openshift-templates/mi-run-template.yaml:38]: This variables are not needed

File: mi-scheduler/base/openshift-templates/mi-run-template.yaml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    value: "latest"
+
+
+objects:
+  - apiVersion: argoproj.io/v1alpha1
+    kind: Workflow
+    metadata:
+      generateName: mi-analysis-
+      annotations:
+        thoth-station.ninja/template-version: 0.1.0
```
Comment (harshad16) [inline mi-scheduler/base/openshift-templates/mi-run-template.yaml:47]: please remove this

File: mi-scheduler/base/argo-workflows/mi-workflow-template.yaml [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+          - name: CEPH_SECRET_KEY
+            valueFrom:
+              secretKeyRef:
+                name: thoth
+                key: ceph-secret-key
+          - name: CEPH_KEY_ID
+            valueFrom:
+              secretKeyRef:
+                name: thoth
+                key: ceph-key-id
```
Comment (pacospace) [inline mi-scheduler/base/argo-workflows/mi-workflow-template.yaml:92]: Use `ceph` CM and keys do not need `ceph` prefix anymore :)

File: mi-scheduler/base/openshift-templates/mi-run-template.yaml [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+                arguments:
+                  parameters:
+                    - name: REPOSITORY
+                      value: ${REPOSITORY}
+                    - name: IMAGE_STREAM_TAG
+                      value: ${IMAGE_STREAM_TAG}
+                    - name: IMAGE_STREAM_REGISTRY
+                      value: ${IMAGE_STREAM_REGISTRY}
+                    - name: IMAGE_STREAM_NAMESPACE
+                      value: ${IMAGE_STREAM_NAMESPACE}
```
Comment (harshad16) [inline mi-scheduler/base/openshift-templates/mi-run-template.yaml:44]: are these needed?

File: mi-scheduler/base/argo-workflows/mi-workflow-template.yaml [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+          - name: PYTHONPATH
+            value: "."
+
+    - name: process
+      inputs:
+        parameters:
+          - name: REPOSITORY
+          - name: IMAGE_STREAM_REGISTRY
+          - name: IMAGE_STREAM_NAMESPACE
+          - name: IMAGE_STREAM_TAG
```
Comment (harshad16) [inline mi-scheduler/base/argo-workflows/mi-workflow-template.yaml:98]: are these needed?

File: mi-scheduler/base/argo-workflows/mi-workflow-template.yaml [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+                - name: IMAGE_STREAM_TAG
+                  value: "{{inputs.parameters.IMAGE_STREAM_TAG}}"
+
+    - name: analyse
+      inputs:
+        parameters:
+          - name: REPOSITORY
+          - name: IMAGE_STREAM_REGISTRY
+          - name: IMAGE_STREAM_NAMESPACE
+          - name: IMAGE_STREAM_TAG
```
Comment (harshad16) [inline mi-scheduler/base/argo-workflows/mi-workflow-template.yaml:53]: are these needed?

File: mi-scheduler/base/argo-workflows/mi-workflow-template.yaml [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+          - name: CEPH_SECRET_KEY
+            valueFrom:
+              secretKeyRef:
+                name: thoth
+                key: ceph-secret-key
+          - name: CEPH_KEY_ID
+            valueFrom:
+              secretKeyRef:
+                name: thoth
+                key: ceph-key-id
```
Comment (harshad16) [inline mi-scheduler/base/argo-workflows/mi-workflow-template.yaml:133]:
```suggestion
                name: ceph
                key: key-id
```

File: mi-scheduler/base/argo-workflows/mi-workflow-template.yaml [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+          - name: CEPH_BUCKET_PREFIX
+            valueFrom:
+              configMapKeyRef:
+                name: ceph
+                key: bucket-prefix
+          - name: CEPH_SECRET_KEY
+            valueFrom:
+              secretKeyRef:
+                name: thoth
+                key: ceph-secret-key
```
Comment (harshad16) [inline mi-scheduler/base/argo-workflows/mi-workflow-template.yaml:128]:
```suggestion
                name: ceph
                key: secret-key
```

File: mi-scheduler/base/argo-workflows/mi-workflow-template.yaml [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
```
Comment (harshad16) [inline mi-scheduler/base/argo-workflows/mi-workflow-template.yaml:83]:
```suggestion
                name: ceph
                key: secret-key
```

File: mi-scheduler/base/argo-workflows/mi-workflow-template.yaml [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+          - name: CEPH_KEY_ID
+            valueFrom:
+              secretKeyRef:
+                name: thoth
+                key: ceph-key-id
```
Comment (harshad16) [inline mi-scheduler/base/argo-workflows/mi-workflow-template.yaml:88]:
```suggestion
                name: ceph
                key: key-id
```

File: mi-scheduler/base/openshift-templates/mi-run-template.yaml (hunk 9 above)
Comment (xtuchyna) [reply mi-scheduler/base/openshift-templates/mi-run-template.yaml:44]: that's right, they are not needed now

File: mi-scheduler/base/argo-workflows/mi-workflow-template.yaml [hunk 16]
```diff
@@ -0,0 +1,114 @@
...
+              parameters:
+                - name: REPOSITORY
+                  value: "{{inputs.parameters.REPOSITORY}}"
+
+    - name: analyse
+      inputs:
+        parameters:
+          - name: REPOSITORY
+      container:
+        image: "mi:latest"
```
Comment (pacospace) [inline mi-scheduler/base/argo-workflows/mi-workflow-template.yaml:37]: should be only `mi` as it uses local image in the namespace, also below

**Pull request merged**